testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketConnectionAttemptTest.java

```diff
@@ -96,7 +95,7 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
 
         cb.handler(handler);
         cb.option(ChannelOption.ALLOW_HALF_CLOSURE, halfClosure);
-        ChannelFuture future = cb.connect(NetUtil.LOCALHOST, TestUtils.getFreePort()).awaitUninterruptibly();
+        ChannelFuture future = cb.connect(NetUtil.LOCALHOST, BAD_PORT).awaitUninterruptibly();
```

Comment:
hmmm true... maybe we should just use a port that no-one will use like tcp port 4 which is not assigned:

```
#                 4/tcp    Unassigned
```
This is from /etc/services...

wdyt ?